aplib.net-demo/Assets/Scripts/WFC/GameObjectPlacer.cs

```diff
@@ -122,31 +113,57 @@ public void Initialize()
+        private static GameObject CreateGameObject(string objectName, Transform parent) =>
+            new(objectName) { transform = { parent = parent } };
 
         /// <summary>
-        /// Spawn a rage potion in the given cell.
+        /// Spawn an item in the given cell.
         /// </summary>
-        /// <param name="cell">The cell to spawn the rage potion in.</param>
-        /// <param name="parent">The parent of the rage potion.</param>
-        public void PlaceRagePotion(Cell cell, Transform parent) =>
-            Instantiate(_ragePotionPrefab, CentreOfCell(cell) + _floorOffset, Quaternion.identity, parent);
+        /// <param name="prefab">The item prefab to spawn.</param>
+        /// <param name="cell">The cell to spawn the item in.</param>
+        /// <param name="parent">The parent of the item.</param>
+        private void PlaceItem(GameObject prefab, Cell cell, Transform parent) =>
+            Instantiate(prefab, CenterOfCell(cell) + _floorOffset, Quaternion.identity, parent);
 
         /// <summary>
-        /// Spawn an ammunition in the given cell.
+        /// Spawns all items in the world.
         /// </summary>
-        /// <param name="cell">The cell to spawn the ammunition in.</param>
-        /// <param name="parent">The parent of the ammunition.</param>
-        public void PlaceAmmunition(Cell cell, Transform parent) =>
-            Instantiate(_ammunitionPrefab, CentreOfCell(cell) + _floorOffset, Quaternion.identity, parent);
+        /// <param name="cells">The cells to spawn the items in.</param>
+        /// <param name="random">The random number generator to use.</param>
+        /// <exception cref="UnityException">Thrown when there are not enough empty cells to place all items.</exception>
+        public void SpawnItems(List<Cell> cells, Random random)
+        {
+            if (_spawnAbleItems.SpawnableItemsCount.Aggregate((x, y) => x + y) > cells.Count)
+                throw new UnityException("Not enough empty cells to place all items.");
+
+            if (_spawnAbleItems.SpawnableItems.Count != _spawnAbleItems.SpawnableItemsCount.Count)
+                throw new UnityException("The number of spawnable items and their counts do not match.");
+
+            GameObject items = CreateGameObject("Items", transform);
+
+            for (int i = 0; i < _spawnAbleItems.SpawnableItems.Count; i++)
+            {
+                GameObject spawnableItem = _spawnAbleItems.SpawnableItems[i];
+
+                GameObject itemParent = CreateGameObject(spawnableItem.name, items.transform);
+
+                for (int j = 0; j < _spawnAbleItems.SpawnableItemsCount[i]; j++)
+                {
+                    Cell cell = cells[random.Next(cells.Count)];
+                    PlaceItem(spawnableItem, cell, itemParent.transform);
+                    cell.CannotAddItem = true;
+                    cells.Remove(cell);
+                }
+            }
```

Comment:
Code to make the previous suggestion work (I wanted to double check if it worked, so I'll just put this here for if you decide on implementing it)
```suggestion
            if (_spawnableItems.Items.Select(x => x.Count).Aggregate((x, y) => x + y) > cells.Count)
                throw new UnityException("Not enough empty cells to place all items.");

            GameObject items = CreateGameObject("Items", transform);

            for (int i = 0; i < _spawnableItems.Items.Count; i++)
            {
                SpawnAbleItems.SpawnableItem spawnableItem = _spawnableItems.Items[i];

                GameObject itemParent = CreateGameObject(spawnableItem.Item.name, items.transform);

                for (int j = 0; j < spawnableItem.Count; j++)
                {
                    Cell cell = cells[random.Next(cells.Count)];
                    PlaceItem(spawnableItem.Item, cell, itemParent.transform);
                    cell.CannotAddItem = true;
                    cells.Remove(cell);
                }
            }
```